thesis-de/content-de.tex

```diff
@@ -2539,7 +2539,229 @@ \section{Prototyp 2}
+Datensatz übermittelt dieser auch die Zieldatenbank und Zieltabelle, sodass
+der Lade-Algorithmus nun nur noch die Daten in das entsprechende Objekt
+einfügen muss. Dabei wird speziell auf die Auslastung des Zielobjektes
+geachtet. Ist die Tabelle zum Beispeil durch einen anderen Prozess belegt,
+wird dies von dem Lade-Algorithmus erkannt und das Einfügen der Daten
+hinten angestellt. Für diesen Teil besitzt die Postgres Datenbank bereits
+vorhandene Frameworks, welche von dem Projektteam genutzt werden. Somit muss
+das Management der Prozessabarbeitung innerhalb des Datenhaltsungssystem nicht
+selber entwickelt, sondern nur eingebunden werden. 
+\nl%
+
+\subsection{Repositorium}
+\label{subsec:postgres_repositorium}
+Der in Punkt~\ref{subsec:postgres_etl} angesprochene Metadatensatz der in
+diesem Projekt für die Metric-Daten jedes virtuellen Servers genutzt wird,
+wurde zusammen mit allen Projektmitgliedern definiert und abgesprochen.
+Dabei wurde sich auf eine spezielle Abspeicherung der Daten geeinigt. Es
+handelt sich um die Verwendung von einer einzelnen Datentabelle, welche
+nur zwei Attribute besitzt. Bei beiden Attributen handelt es sich um ein
+JSON Objekt. Ersteres beinhaltet Metadaten zu dem einzelnen Datensatz.
+Das ist zum Beispiel eine generierte ID, um den Datensatz spezifisch
+selektieren zu können, oder die Informationen seit wann der Datensatz
+in der Datenhaltung existiert, beziehungsweise wann und von welchem System
+angefasst wurde. Zweites JSON Objekt/Attribut beinhaltet den eigentlichen
+Datensatz, also die werthaltigen Informationen, welche auch später analysiert
+werden. Die Inhalte des Attributes werden klar in dem Metadatensatz definiert.
+Der Vorteil dieser Datenspeicherung ist zum einen die Effizienz des
+Speicherplatzes sowie Geschwindigkeit und zum anderen das Zusammenhalten der
+zugehörigen Informationen. Desweiteren können in diesem Model sehr einfach neue
+Informationen hinzugefügt werden, indem diese in dem Metadatensatz definiert
+werden. Dies ist ein Vorteil der Skalierbarkeit und es müssen anders wie bei
+Datenbanken in der dritten Normalform die voneinander abhängigen Daten nicht
+in jeweils eine eigene Tabelle ausgelagert werden.
+
+Da sich bei diesem Model die Geschwindigkeit der Datenabfragen mit der Menge
+der Daten in einer Tabelle verlangsamt, wird die automatische Partitionierung
+angewandt, welche im nachfolgendem Punkt~\ref{subsec:postgres_partition}
+beschrieben wird.
+\nl%
+
+\subsection{Partitionierung}
+\label{subsec:postgres_partition}
+Bei der Partitionierung in dem Postgres-System muss zwischen zwei Arten von
+Partitionierung entschieden werden. Ersteres ist die manuelle Partitionierung
+durch den Administrator. Dabei werden über einen Index Datenbereiche definiert,
+welche logisch voneinander getrennt werden. Mit nachfolgendem Code-Prototyp
+kann solch eine Trennung im groben vom Administrator definiert werden.
+
+\begin{listing}
```

Comment:
das ist ein sehr langes listing, ist schwer das sauber zu positionieren. sollen wir den lieber in den anhang schieben?